contracts/modules/token/TokenController.sol

```diff
@@ -206,21 +207,63 @@ contract TokenController is ITokenController, LockHandler, LegacyMasterAware {
   /// Function used to claim all pending rewards in one tx. It can be used to selectively withdraw
   /// rewards.
   ///
-  /// @param batchSize  The maximum number of iterations to avoid unbounded loops
-  /// @param batchSize  The maximum number of iterations to avoid unbounded loops
+  /// @param forUser          The address for whom the governance and/or assessment rewards are
+  ///                         withdrawn.
+  /// @param fromGovernance   When true, governance rewards are withdrawn.
+  /// @param fromAssessment   When true, assessment rewards are withdrawn.
+  /// @param batchSize        The maximum number of iterations to avoid unbounded loops when
+  ///                         withdrawing governance and/or assessment rewards.
+  /// @param stakingPools     The addresses of the staking pools that have withdrawable rewards.
+  ///                         When empty, no staking rewards are withdrawn.
+  /// @param stakingTokenIds  The ids of the tokens corresponding to every staking pool provided
+  ///                         above which have withdrwable rewards.
+  /// @param stakingTermIds   The ids of the terms corresponding to every token provided above
+  ///                         which have withdrwable rewards.
   function withdrawPendingRewards(
     address forUser,
     bool fromGovernance,
     bool fromAssessment,
-    uint batchSize
+    uint batchSize,
+    address[] calldata stakingPools,
+    uint[][] calldata stakingTokenIds,
+    uint[][][] calldata stakingTermIds
   ) external isMemberAndcheckPause {
     if (fromAssessment) {
       assessment.withdrawRewards(forUser, batchSize.toUint104());
     }
+
     if (fromGovernance) {
       uint governanceRewards = governance.claimReward(forUser, batchSize);
       require(governanceRewards > 0, "TokenController: No withdrawable governance rewards");
-      require(token.transfer(forUser, governanceRewards), "TokenController: Governance rewards transfer failed");
+      require(
+        token.transfer(forUser, governanceRewards),
+        "TokenController: Governance rewards transfer failed"
+      );
+    }
+
+    require(
+      stakingPools.length == stakingTokenIds.length,
+      "TokenController: Provide all withdrawable the token ids for each staking pool"
+    );
+    require(
+      stakingTokenIds.length == stakingTokenIds.length,
+      "TokenController: Provide all expired terms for every token from each staking pool"
+    );
+    for (uint i = 0; i < stakingPools.length; i++) {
+      require(
+        stakingTokenIds[i].length == stakingTermIds[i].length,
+        "TokenController: Provide all expired terms for each token"
```

Comment:
I could, they don't serve any business logic purpose, it just makes it more obvious what is going on when there's a mismatch in the arrays, rather than the cryptic panic code.